src/main/kotlin/mate/academy/service/FigureSupplier.kt

```diff
@@ -0,0 +1,47 @@
+package mate.academy.service
+
+import mate.academy.model.Circle
+import mate.academy.model.Color
+import mate.academy.model.Figure
+import mate.academy.model.FigureIndex
+import mate.academy.model.IsoscelesTrapezoid
+import mate.academy.model.Rectangle
+import mate.academy.model.RightTriangle
+import mate.academy.model.Square
+import kotlin.random.Random
+
+class FigureSupplier {
+    private val colorSupplier = ColorSupplier()
+
+    companion object {
+        const val DEFAULT_RADIUS = 10.0
+        const val MAX_RANDOM_VALUE = 8.0
+    }
+
+    fun getRandomFigure(): Figure {
+        return when (Random.nextInt(FigureIndex.values().size)) {
```

Comment:
For simplifying you may use kotlin's random() extension method instead.
Smth like that
YOUR_ENUM.values().random()

In this case you don't need to use indexes for figures